pulsar-broker/src/main/java/org/apache/pulsar/broker/service/PulsarCommandSenderImpl.java

```diff
@@ -243,8 +243,13 @@ public ChannelPromise sendMessagesToConsumer(long consumerId, String topicName,
                 // increment ref-count of data and release at the end of process:
                 // so, we can get chance to call entry.release
                 metadataAndPayload.retain();
-                // skip raw message metadata since broker timestamp only used in broker side
-                Commands.skipBrokerEntryMetadataIfExist(metadataAndPayload);
+                // skip broker entry metadata if consumer-client doesn't support broker entry metadata or the
+                // features is not enabled
+                if (cnx.getRemoteEndpointProtocolVersion() < ProtocolVersion.v18.getValue()
```

Comment:
It is better to use the `FeatureFlag` mechanism instead of checking the `ProtocolVersion`. This will ensure clients can phase in with the support of this feature.

Reply:
that's interesting. I wasn't aware of that mechanism

Reply:
Make sense, even if the client with `ProtocolVersion.v18` but does not support get broker entry metadata, the broker can skip the broker entry metadata to reduce the network workload.

Reply:
> It is better to use the `FeatureFlag` mechanism instead of checking the `ProtocolVersion`. This will ensure clients can phase in with the support of this feature.

Thanks for your suggestion. Maybe it's better to check the `FeatureFlag` and the `ProtocolVersion` Both?

Reply:
> It is better to use the `FeatureFlag` mechanism instead of checking the `ProtocolVersion`. This will ensure clients can phase in with the support of this feature.

PIP-70 adds a feature flag `supports_broker_entry_metadata`, so I use it to check here. Can you review it again?